deepcell_tracking/utils.py

```diff
@@ -200,26 +203,86 @@ def trk_folder_to_trks(dirname, trks_filename):
 
 
 def save_trks(filename, lineages, raw, tracked):
-    """Saves raw, tracked, and lineage data into one trks_file.
+    """Saves raw, tracked, and lineage data from multiple movies into one trks_file.
 
     Args:
-        filename (str): full path to the final trk files.
-        lineages (dict): a list of dictionaries saved as a json.
+        filename (str or io.BytesIO): full path to the final trk files or bytes object
+            to save the data to
+        lineages (list): a list of dictionaries saved as a json.
         raw (np.array): raw images data.
         tracked (np.array): annotated image data.
 
     Raises:
         ValueError: filename does not end in ".trks".
     """
-    if not str(filename).lower().endswith('.trks'):
+    ext = os.path.splitext(str(filename))[-1]
+    if not isinstance(filename, io.BytesIO) and ext != '.trks':
         raise ValueError('filename must end with `.trks`. Found %s' % filename)
 
-    with tarfile.open(filename, 'w:gz') as trks:
+    save_track_data(filename=filename,
+                    lineages=lineages,
+                    raw=raw,
+                    tracked=tracked,
+                    lineage_name='lineages.json')
+
+
+def save_trk(filename, lineages, raw, tracked):
+    """Saves raw, tracked, and lineage data for one movie into a trk_file.
+
+    Args:
+        filename (str or io.BytesIO): full path to the final trk files or bytes
+            object to save the data to
+        lineages (list or dict): a list of a single dictionary or a single
+            lineage dictionary
+        raw (np.array): raw images data.
+        tracked (np.array): annotated image data.
+
+    Raises:
+        ValueError: filename does not end in ".trks".
+    """
+    ext = os.path.splitext(str(filename))[-1]
+    if not isinstance(filename, io.BytesIO) and ext != '.trk':
+        raise ValueError('filename must end with `.trk`. Found %s' % filename)
+
+    # Check that lineages is a dictionary or list of length 1
+    if isinstance(lineages, list):
+        if len(lineages) > 1:
+            raise ValueError('For trk file, lineages must be a dictionary '
+                             'or list with a single dictionary')
+        else:
+            lineages = lineages[0]
+
+    save_track_data(filename=filename,
+                    lineages=lineages,
+                    raw=raw,
+                    tracked=tracked,
+                    lineage_name='lineage.json')
+
+
+def save_track_data(filename, lineages, raw, tracked, lineage_name):
```

Comment:
To DRY out the repo and prevent multiple minor revs, could we change `.dump` in the `CellTracker` class to make use of this function?

Reply:
Yes please, let's call our new code in `.dump`.